xgi/linalg/matrix.py

```diff
@@ -314,13 +321,12 @@ def multiorder_laplacian(H, orders, weights, rescale_per_node=False, index=False
             L_multi += L * w / np.mean(K)
 
     if index:
-        _, rowdict, _ = incidence_matrix(H, index=True)
-        return L_multi, rowdict
+        return L_multi, dict(zip(range(H.num_nodes), H.nodes))
```

Comment:
Why not use index=True anymore? This won't work with nodes that are not ordered?

Reply:
I essentially copied the lines in the `incidence_matrix` function generating the `nodedict` and `edgedict` dictionaries because most of the incidence matrix function was unnecessary here. But not sure if that was your actual question...

Reply:
Oh I see good call then